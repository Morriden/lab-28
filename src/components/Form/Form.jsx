import React from 'react';
import PropTypes from 'prop-types';


const Form = ({ linkInput, bodyInput, onLinkChange, onBodyChange, onSubmit, handleOptionChange, method }) => (
  <form onSubmit={onSubmit}>
    <input type="text" value={linkInput} onChange={onLinkChange} />
    <input type="text" value={bodyInput} onChange={onBodyChange} />
    <input type="radio" value="GET" name="route" onChange={handleOptionChange} checked={method === 'GET'}/>
    <input type="radio" value="POST" name="route" onChange={handleOptionChange} checked={method === 'POST'}/>
    <input type="radio" value="PUT" name="route" onChange={handleOptionChange} checked={method === 'PUT'}/>
    <input type="radio" value="DELETE" name="route" onChange={handleOptionChange} checked={method === 'DELETE'}/>
    <button>Submit</button>
  </form>
);

Form.propTypes = {
  linkInput: PropTypes.string.isRequired,
  onLinkChange: PropTypes.func.isRequired,
  onBodyChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  bodyInput: PropTypes.string.isRequired,
  handleOptionChange: PropTypes.func.isRequired
};

export default Form;
