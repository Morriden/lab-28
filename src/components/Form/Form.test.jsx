import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  it('should render to input fields and one button for now.', () => {
    const wrapper = shallow(
    <Form 
        linkInput="somewebsite.com"
        bodyInput="someJSONbodyinfo"
        onLinkChange={() => {}}
        onBodyChange={() => {}}
        onSubmit={() => {}}
        handleOptionChange={() => {}}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});