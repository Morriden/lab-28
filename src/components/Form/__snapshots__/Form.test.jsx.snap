// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component should render to input fields and one button for now. 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="somewebsite.com"
  />
  <input
    onChange={[Function]}
    type="text"
    value="someJSONbodyinfo"
  />
  <input
    checked={false}
    name="route"
    onChange={[Function]}
    type="radio"
    value="GET"
  />
  <input
    checked={false}
    name="route"
    onChange={[Function]}
    type="radio"
    value="POST"
  />
  <input
    checked={false}
    name="route"
    onChange={[Function]}
    type="radio"
    value="PUT"
  />
  <input
    checked={false}
    name="route"
    onChange={[Function]}
    type="radio"
    value="DELETE"
  />
  <button>
    Submit
  </button>
</form>
`;
